fix: move etcd to `cri` containerd runner

smira opened the pull request:
This fixes a problem when Talos pulls `etcd` image one every reboot, as
`etcd` was running in the system containerd which is completely
ephemeral (backed by `tmpfs`).

Also skip pulling if image is already present and unpacked (same fix for
the `kubelet` image).

Fixes #3229

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

smira commented:
/approve

andrewrynhard commented on internal/app/machined/pkg/system/services/kubelet.go:
```diff
@@ -112,9 +114,25 @@ func (k *Kubelet) PreFunc(ctx context.Context, r runtime.Runtime) error {
 	// Pull the image and unpack it.
 	containerdctx := namespaces.WithNamespace(ctx, "k8s.io")
 
-	_, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Machine().Kubelet().Image())
+	var pullRequired bool
+
+	img, err := client.GetImage(containerdctx, r.Config().Machine().Kubelet().Image())
 	if err != nil {
-		return err
+		pullRequired = true
+	} else {
+		var unpacked bool
+
+		unpacked, err = img.IsUnpacked(containerdctx, "")
+		if err != nil || !unpacked {
+			pullRequired = true
+		}
+	}
+
+	if pullRequired {
+		_, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Machine().Kubelet().Image())
+		if err != nil {
+			return err
+		}
```
Worth putting this in a common place? Like our `containerd` package.

smira replied:
yep, I think that makes sense. probably even `image.Pull` function itself with an option?

andrewrynhard replied:
Makes sense.

andrewrynhard reviewed (approved):
I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.

smira commented:
> I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.

yeah, we could... `talosctl containers -k` now doesn't see these, as they're not launched via CRI, but we could add even more separation here

smira commented:
e

> > I wonder if we should have etcd and kubelet be in the `system` namespace (but still in the CRI containerd). Makes for a cleaner separation. Also, I left a question. I know we need to get just get _something_ so don't let me shoe you down.
> 
> yeah, we could... `talosctl containers -k` now doesn't see these, as they're not launched via CRI, but we could add even more separation here

moved to system namespace and refactored to reduce duplication
